substratevm/src/com.oracle.svm.core.jdk15/src/com/oracle/svm/core/jdk15/Target_java_lang_invoke_MethodHandles_JDK15OrLater.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2020, 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.jdk15;
+
+import java.lang.invoke.MethodHandles;
+// Checkstyle: stop
+import sun.invoke.util.VerifyAccess;
+// Checkstyle: resume
+import jdk.internal.misc.Unsafe;
+
+import com.oracle.svm.core.jdk.JDK15OrLater;
+import com.oracle.svm.core.SubstrateUtil;
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "java.lang.invoke.MethodHandles", innerClass = "Lookup", onlyWith = JDK15OrLater.class)
```

Comment:
I would like to have this substitution only for exactly JDK 15, and a proper fix merged into JDK 16 at the same time we merge this PR. It is not sustainable to have substitutions like this, so we need to adopt a "fix properly in master first" approach.

Reply:
I've restricted the change to JDK15 and initiated a dialog with the JDK core libraries team.

Reply:
Pushed a change to JDK16 and restricted this substitution to JDK15.